src/otg/dataset/study_index.py

```diff
@@ -477,3 +477,113 @@ def from_source(
                 )
             )
         )
+
+
+@dataclass
+class StudyIndexUKBiobank(StudyIndex):
+    """Study index dataset from UKBiobank.
+
+    The following information is extracted:
+
+    - studyId
+    - pubmedId
+    - publicationDate
+    - publicationJournal
+    - publicationTitle
+    - publicationFirstAuthor
+    - traitFromSource
+    - ancestry_discoverySamples
+    - ancestry_replicationSamples
+    - initialSampleSize
+    - nCases
+    - replicationSamples
+
+    Some fields are populated as constants, such as projectID, studyType, and initial sample size.
+    """
+
+    @classmethod
+    def from_source(
+        cls: type[StudyIndexUKBiobank],
+        ukbiobank_studies: DataFrame,
+    ) -> StudyIndexUKBiobank:
+        """This function ingests study level metadata from UKBiobank.
+
+        Args:
+            ukbiobank_studies (DataFrame): UKBiobank study manifest file loaded in spark session.
+
+        Returns:
+            StudyIndexUKBiobank: Annotated UKBiobank study table.
+        """
+        return cls(
+            _df=(
+                ukbiobank_studies.select(
+                    f.col("code").alias("studyId"),
+                    f.col("trait").alias("traitFromSource"),
+                    f.col("n_cases").cast("long").alias("nCases"),
+                    f.col("n_total").cast("string").alias("initialSampleSize"),
+                    f.col("in_path").alias("summarystatsLocation"),
+                )
+                # Replace any trait double spaces with single.
+                .withColumn(
+                    "traitFromSource",
+                    f.regexp_replace(f.col("traitFromSource"), r" +", " "),
```

Comment:
I'm not sure what this function does. As I understand, it removes any extra whitespace, however the data looks the same to me after applying it. Am I understanding things correctly?

Reply:
It should be removing whitespace, I was trying to bring over the same function from the old make_UKB_study_table.py [here](https://github.com/opentargets/genetics-v2d-data/blob/4b5e6f9989dca73f93dcdeaa8fc8d1bd3f51690f/scripts/make_UKB_study_table.py#L90) - but I can go ahead and remove this if it's not needed

Reply:
I don't think it is, tbf